### PR TITLE
Browserstack observability reload session fix

### PR DESCRIPTION
**Pull request by rev-doshi**
## Proposed changes
Fix for sending correct session id to Browserstack Observability in case of reloadSession

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Polish (an improvement for an existing feature)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (improvements to the project's docs)
- [ ] Specification changes (updates to WebDriver command specifications)
- [ ] Internal updates (everything related to internal scripts, governance documentation and CI files)

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [ ] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Backport Request

[//]: # (The current `main` branch is the development branch for WebdriverIO v9. If your change should be released to the current major version of WebdriverIO (v8), please raise another PR with the same changes against the `v8` branch.)

- [ ] This change is solely for `v9` and doesn't need to be back-ported
- [ ] Back-ported PR at #12368

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

**Review by 07souravkunda — APPROVED** [review]

File: packages/wdio-browserstack-service/src/insights-handler.ts [hunk 1]
```diff
@@ -864,12 +864,15 @@ class _InsightsHandler {
...
+        const caps = (this._browser as WebdriverIO.Browser)?.capabilities as WebdriverIO.Capabilities
+        const sessionId = (this._browser as WebdriverIO.Browser)?.sessionId
+
         return {
-            capabilities: this._platformMeta?.caps,
-            session_id: this._platformMeta?.sessionId,
-            browser: this._platformMeta?.browserName,
-            browser_version: this._platformMeta?.browserVersion,
-            platform: this._platformMeta?.platformName,
+            capabilities: caps,
```
Comment (karanshah-browserstack) [inline packages/wdio-browserstack-service/src/insights-handler.ts:871]: Do we have existing test cases to cover this?

File: packages/wdio-browserstack-service/src/insights-handler.ts [hunk 2]
```diff
@@ -864,12 +864,15 @@ class _InsightsHandler {
...
+            session_id: sessionId,
```
Comment (karanshah-browserstack) [inline packages/wdio-browserstack-service/src/insights-handler.ts:872]: Please ensure to raise this on v7 too and check it works for w3c and non w3c mode.

Comment (sriteja777) [reply packages/wdio-browserstack-service/src/insights-handler.ts:872]: Have created [V8](https://github.com/webdriverio/webdriverio/pull/12368) and [V7](https://github.com/webdriverio/webdriverio/pull/12369) PRs

Comment (sriteja777) [reply packages/wdio-browserstack-service/src/insights-handler.ts:872]: The code that is changed is already present, so it's already working for W3c and non W3c

**Review by karanshah-browserstack — APPROVED** [review]

File: packages/wdio-browserstack-service/src/insights-handler.ts (hunk 1 above)
Comment (rev-doshi) [reply packages/wdio-browserstack-service/src/insights-handler.ts:871]: We didn't have existing cases, have added them now.

**Comment by christian-bromann** [comment]
> Do you have a branch for backporting this to `v8` (or `v7`)?

Just reviewed and merged them.

**Pull request merged**